test/linalg/lu.jl

```diff
@@ -83,10 +83,29 @@ debug && println("(Automatic) Square LU decomposition. eltya: $eltya, eltyb: $el
                 @test norm(a*(lua\c) - c, 1) < ε*κ*n # c is a vector
                 @test norm(a'*(lua'\c) - c, 1) < ε*κ*n # c is a vector
                 @test AbstractArray(lua) ≈ a
-                if eltya <: Real && eltyb <: Real
```

Comment:
`'` and `.'` are equivalent for real matrices but not for complex ones, and I think the idea has been to restrict the `.'` test to complex matrices. But then it should have read `<: Complex`. When fixing that I thought it wasn't really necessary to do that test conditionally; makes it easier to read what's happening.